dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -646,7 +646,7 @@ void ExpressionAnalyzer::findExternalTables(ASTPtr & ast)
     StoragePtr external_storage;
 
     if (ASTIdentifier * node = typeid_cast<ASTIdentifier *>(ast.get()))
-        if (node->kind == ASTIdentifier::Table)
+        if (node->special())
```

Comment:
The only dangerous place. It will check databases and formats in  tryGetExternalTable()